File: gradle-plugin/src/main/java/org/fairy/gradle/PlatformType.java
package org.fairy.gradle;

import org.fairy.gradle.file.FileGenerator;
import org.fairy.gradle.file.FileGeneratorBukkit;

public enum PlatformType {

    BUKKIT {
        @Override
        public FileGenerator createFileGenerator() {
            return new FileGeneratorBukkit();
        }
    },
    BUNGEE,
    VELOCITY,
    NUKKIT,
    INDEPENDENT;

    public String getDependencyName() {
        return this.name().toLowerCase();
    }

    public FileGenerator createFileGenerator() {
        throw new UnsupportedOperationException("Platform " + this.name() + " hasn't been supported.");
    }

}

File: gradle-plugin/src/main/kotlin/io/fairyproject/gradle/platform/PlatformType.kt
package io.fairyproject.gradle.platform

/**
 * The platform type.
 */
enum class PlatformType {
    BUKKIT,
    APP,
    CORE;

    val dependencyName: String
        get() = name.lowercase()

    companion object {
        val VALUES = values().toList()
    }
}